zipline/finance/blotter/core.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from functools import partial
+
+from zipline.finance.blotter.blotter import Blotter
+from zipline.utils.compat import mappingproxy
+
+
+def _make_blotters_core():
```

Comment:
This looks like it's copying the pattern of the `make_bundle_core` functionality, so this is probably more of a question for @llllllllll than for @jnazaren, but is there a reason we're using this odd javascript-style closure construct instead of just defining a class?

Reply:
An alternative (and I would argue, much clearer and simpler) way to build a similar registration API would be to have a global instance of something like a `BlotterRegistry` class.  For example, we do something analogous for trading calendars here: https://github.com/quantopian/zipline/blob/d4de30860b15e18aa5814167f8bceb98a71a15b5/zipline/utils/calendars/calendar_utils.py#L39-L267

Reply:
Originally it was just a few functions in a module with a global dict, and then I wanted to test it in an isolated way so I just copied them into a closure. Making it a class with methods is fine, but I don't have a strong opinion on this. If you have even a slight preference it's fine to change it.

Reply:
I'm definitely on board with encapsulating the global state in a way that makes isolated testing possible. My thought here is just that having a closure that defines a dictionary and a bunch of functions that share the state of that dictionary is essentially a roundabout way of creating an instance of a class.

Reply:
@jnazaren Can you do this?

Reply:
Ok, will this need a 'factories' storage like the one with calendars for lazy instantiation?